src/common/reducers/entities/index.js

```diff
@@ -1,26 +1,22 @@
 import merge from 'lodash/merge';
 
 import * as RepoActionTypes from '../../actions/repository';
-
 import repository from './repository';
 
-
 // TODO snaps
 
-export function entities(state = { snaps: {}, repos: {} }, action) {
+export function entities(state = {
+  snaps: {},
+  repos: {},
+  owners: {}
+}, action) {
   if (action.payload && action.payload.entities) {
     return merge({}, state, action.payload.entities);
   }
 
   // only modify repos if action is one of REPO_ types
-  if (action.payload && action.payload.id && RepoActionTypes[action.type]) {
-    return {
-      ...state,
-      repos: {
-        ...state.repos,
-        [action.payload.id]: repository(state.repos[action.payload.id], action)
-      }
-    };
+  if (RepoActionTypes[action.type]) {
+    return repository(state, action);
```

Comment:
I don't like that `repository` actually returns whole entities object.

Maybe renaming it to `repositoryEntities` would make it more clear, but still, it's a bit confusing.

I think repository helper/reducer shouldn't know about entities structure, so maybe something like:

```js
return {
   ...state,
   repos: reposEntities(state.repos, action)
}
```

This way `reposEntities` doesn't need to know about structure of `entities`, only about repos object inside.
And probably it may help us move away from checking action types in `entities` at all, as it's `reposEntities` that may check it and decide weather repos need to be touched at all.